clinica/pipelines/deeplearning_prepare_data/deeplearning_prepare_data_utils.py

```diff
@@ -0,0 +1,218 @@
+
+    else:
+
+        # axial
+        slice_list_axi = range(M, image_tensor.shape[2] - N)  # delete the first M slices and last N slices
+        for index_slice, index_slice_list in zip(slice_list_axi, range(len(slice_list_axi))):
+            # for i in slice_list:
+            # sagital
+            slice_select_axi = image_tensor[:, :, index_slice]
+
+            # convert the slices to images based on if transfer learning or not
+            # train from scratch
+            extracted_slice_original_axi = slice_select_axi.unsqueeze(0)  # shape should be 1 * W * L
+
+            # train for transfer learning, creating the fake RGB image.
+            slice_select_axi = (slice_select_axi - slice_select_axi.min()) / (slice_select_axi.max() - slice_select_axi.min())
+            extracted_slice_rgb_axi = torch.stack((slice_select_axi, slice_select_axi, slice_select_axi))  # shape should be 3 * W * L
+
+            # save into .pt format
+            if slice_mode == 'original':
+                output_file_original.append(
+                        os.path.join(
+                            basedir,
+                            os.path.basename(preprocessed_T1).split('.pt')[0]
+                            + '_axis-axi_originalslice-'
+                            + str(index_slice)
+                            + '.pt'
+                            )
+                        )
+                torch.save(extracted_slice_original_axi.clone(), output_file_original[index_slice_list])
+            elif slice_mode == 'rgb':
+                output_file_rgb.append(
+                        os.path.join(
+                            basedir,
+                            os.path.basename(preprocessed_T1).split('.pt')[0]
+                            + '_axis-axi_rgbslice-'
+                            + str(index_slice)
+                            + '.pt'
+                            )
+                        )
+                torch.save(extracted_slice_rgb_axi.clone(), output_file_rgb[index_slice_list])
+
+    return output_file_rgb, output_file_original
+
+
+def extract_patches(preprocessed_T1, patch_size, stride_size):
+    """This function extracts the patches from three directions
+
+    Args:
+        preprocessed_T1: tensor version of the nifty MRI
+        patch_size: size of each patch
+        stride_size: size of each stride
+
+    Returns:
+        file: file stored in the disk
+    """
+    import torch
+    import os
+
+    basedir = os.getcwd()
+    image_tensor = torch.load(preprocessed_T1)
+
+    # use classifiers tensor.upfold to crop the patch.
+    patches_tensor = image_tensor.unfold(1, patch_size, stride_size).unfold(2, patch_size, stride_size).unfold(3, patch_size, stride_size).contiguous()
+    # the dimension of patch_tensor should be [1, patch_num1, patch_num2, patch_num3, patch_size1, patch_size2, patch_size3]
+    patches_tensor = patches_tensor.view(-1, patch_size, patch_size, patch_size)
```

Comment:
What does the `-1` do in `patches_tensor.view(-1 ...` ?

Reply:
> What does the `-1` do in `patches_tensor.view(-1 ...` ?

In that case `-1` the new size of the tensor is inferred automatically from the original dimensions.